src/makeInfinityAdder.js

```diff
@@ -6,7 +6,29 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  return (a) => {
+    let sum = a;
+
+    if (!a) {
+      return 0;
+    }
+
+    const getSum = (b) => {
+      if (b >= 0) {
```

Comment:
But what if we want to add negative numbers? Your function will just return total and reset sum to zero.